### PR TITLE
Add caching url test

### PR DESCRIPTION
**Pull request by octonawish-akcodes**
Need reviews @spbnick

### CODE REVIEW

File: test_main.py [hunk 1]
```diff
@@ -37,3 +39,31 @@ def test_import():
...
         os.environ.update(orig_env)
+
+def test_url_caching(empty_deployment):
+    publisher = kcidb.mq.URLListPublisher(
+        os.environ["PROJECT"],
+        os.environ["KCIDB_UPDATED_URLS_TOPIC"]
+    )
+    publisher.publish([
+        "https://github.com/kernelci/kcidb/blob/main/test_kcidb.py",
+        "https://github.com/kernelci/kcidb/blob/main/get-url-stats"
```
Comment (spbnick) [inline test_main.py:50]: Use the URLs of the "raw" files. We would get a smaller and faster response, and they would cost GitHub much less.
E.g. the first one should be [https://raw.githubusercontent.com/kernelci/kcidb/main/test_kcidb.py](https://raw.githubusercontent.com/kernelci/kcidb/main/test_kcidb.py)

File: test_main.py [hunk 2]
```diff
@@ -37,3 +39,31 @@ def test_import():
     finally:
         os.environ.clear()
         os.environ.update(orig_env)
+
+def test_url_caching(empty_deployment):
+    publisher = kcidb.mq.URLListPublisher(
+        os.environ["PROJECT"],
```
Comment (spbnick) [inline test_main.py:45]: Uh, this one should be `GCP_PROJECT`, shouldn't it?

**Review by spbnick — CHANGES_REQUESTED** [review]
Looks pretty good! Go ahead and make it work now!

Please separate the addition of the `URLListPublisher` into its own commit, though.

File: test_main.py (hunk 2 above)
Comment (octonawish-akcodes) [reply test_main.py]: Yeah it was just a demo layout to show u. I did changed it.

File: test_main.py (hunk 1 above)
Comment (octonawish-akcodes) [reply test_main.py]: got it

**Pull request closed without merging**